### ISSUE TITLE
oshmem_info link failure for 4.0.5

### ISSUE THREAD

**Issue by hoba87** [open-mpi/ompi#8106]
## Background information
I get undefined reference for oshmem_info:
```
  CC       oshmem_info.o
  CC       param.o
  GENERATE oshmem_info.1
  CCLD     oshmem_info
../../../oshmem/.libs/liboshmem.so: undefined reference to `orte_process_info'
../../../oshmem/.libs/liboshmem.so: undefined reference to `orte_errmgr'
../../../oshmem/.libs/liboshmem.so: undefined reference to `orte_namelist_t_class'
../../../oshmem/.libs/liboshmem.so: undefined reference to `orte_initialized'
../../../oshmem/.libs/liboshmem.so: undefined reference to `orte_util_print_name_args'
../../../oshmem/.libs/liboshmem.so: undefined reference to `ORTE_ERROR_LOG'
../../../oshmem/.libs/liboshmem.so: undefined reference to `orte_show_help'
../../../oshmem/.libs/liboshmem.so: undefined reference to `orte_show_help_is_available'
```

### What version of Open MPI are you using? (e.g., v3.0.5, v4.0.2, git branch name and hash, etc.)
4.0.5

### Describe how Open MPI was installed (e.g., from a source/distribution tarball, from a git clone, from an operating system distribution package, etc.)
I use the source tarball and configure with:
./configure --prefix=/opt/openmpi/openmpi-4.0.5/ --with-verbs --with-pmix=/opt/pmix/pmix-3.1.5 --with-ompi-pmix-rte --with-cuda=/opt/nvidia/cuda-10.1 2>&1 | tee config.log

### Please describe the system on which you are running
* Operating system/version: CentOS 8
* Computer hardware: Intel Xeon E5 v4
* Network type: Ethernet, Infiniband

-----------------------------

## Details of the problem
I get several warnings, but no errors:
```
configure: WARNING:  -fno-strict-aliasing has been added to CFLAGS
configure: WARNING:  -finline-functions has been added to CXXFLAGS
configure: WARNING:  -finline-functions has been added to CXXFLAGS
configure: WARNING: *** Did not find corresponding C type
configure: WARNING: Compiler INTEGER*16 and  mismatch; MPI datatype unsupported
configure: WARNING: external hwloc version is less than internal version 2.0
configure: WARNING: using internal hwloc
configure: WARNING: "-fvisibility=hidden" has been added to the hwloc CFLAGS
configure: WARNING: valgrind.h not found
configure: WARNING: Cannot compile this component
configure: WARNING:  -fno-strict-aliasing has been added to CFLAGS
configure: WARNING: Unknown architecture ... proceeding anyway
configure: WARNING: File locks may not work with NFS.  See the Installation and
configure:26308: WARNING:  -finline-functions has been added to CXXFLAGS
configure:27984: WARNING:  -finline-functions has been added to CXXFLAGS
configure:55240: WARNING: *** Did not find corresponding C type
configure:55694: WARNING: Compiler INTEGER*16 and  mismatch; MPI datatype unsupported
configure:81101: WARNING: external hwloc version is less than internal version 2.0
configure:81103: WARNING: using internal hwloc
configure:86106: WARNING: "-fvisibility=hidden" has been added to the hwloc CFLAGS
configure:149259: WARNING: valgrind.h not found
configure:149261: WARNING: Cannot compile this component
```

**Comment by hoba87** [comment]
I have realized, that it works without option --with-ompi-pmix-rte and I should use --with-ucx instead of --with-verbs. However I leave the decision to close the issue to the developers.

**Comment by hppritcha** [comment]
This is an issue in the oshmem component of Open MPI.  oshmem should not have a direct dependency on orte.

**Comment by hppritcha** [comment]
@hoopoepg Sergey could you check on this?

**Comment by jsquyres** [comment]
FYI: @open-mpi/ucx this is apparently still an issue.

**Comment by hoopoepg** [comment]
yes, working over this issue

**Comment by hoopoepg** [comment]
hi @hoba87

thank you for bug report!!!
I tried to reproduce this issue using default CentOS 8 environment but without success.

as I can see you used non-default PMIX library, what is configure string for PMIX?
what else non-default dependencies were used?
could you also check is issue is actual for later OMPI? 4.1 and 5.0

thank you again

**Comment by hoopoepg** [comment]
and what compiler was used? default from OS?

**Comment by hppritcha** [comment]
i was using gcc 9 or something like that.  the problematic thing is the --with-ompi-pmix-rte configure option.  if you don't use this you won't see the problem.

**Comment by rhc54** [comment]
Yes - that option turns off the building of ORTE, and so anything that depends on ORTE will break.

**Comment by hoba87** [comment]
hi @hoopoepg,
the configure string for PMIX was just `./configure --prefix=...`, so nothing specific besides the install directory here. I used the default compiler, that is gcc 8.3.1.
For openmpi 4.1.0 with the configure and compile strings:
`./configure --prefix=/opt/openmpi/openmpi-4.1.0/ --with-verbs --with-pmix=/opt/pmix/pmix-3.1.5 --with-ompi-pmix-rte > config.log 2> config.err
CPPFLAGS="-I/opt/pmix/pmix-3.1.5/include" make -j20 | tee make.log`
I get an error during compile time:
> mtl_ofi_component.c: In function 'select_ofi_provider':
> mtl_ofi_component.c:362:69: error: 'pmix_process_info_t' {aka 'struct <anonymous>'} has no member named 'cpuset'
>                                                    ompi_process_info.cpuset,

**Comment by hppritcha** [comment]
@open-mpi/ofi  this looks significant

**Comment by gpaulsen** [comment]
Thanks Howard.

**Comment by janjust** [comment]
I've closed the corresponding PRs due to concerns over side-effects this may bring into release branches.
